lint

```diff
@@ -171,6 +171,27 @@ lint_files() {
     exit $lint_result
 }
 
+filter_out() {
+    local globs="$1"
+    if [ -n "$globs" -a -r "$globs" ]; then
+        # Files to debug the regular expressions, as they get quite complex:
+        local files_in="$(mktemp /tmp/lint_files_in_XXX.txt)"
+        local files_out="$(mktemp /tmp/lint_files_out_XXX.txt)"
+        # Globs blacklist found: filter out stdin before passing it to stdout:
+        while read -r filename; do
+            # Remove comments. Remove empty lines. Expand directories (e.g. "bin/" -> "bin/.*"). And then regexify globs.
+            # These regexes are then passed to grep to filter out matching elements from stdin:
+            echo "${filename}" | tee -a "$files_in" | grep -vEf <(sed '/^#.*$/d ; /^\s*$/d ; s/^\(.*\/\)$/\1*/g ; s/\([.|]\)/\\\1/g ; s/\?/./g ; s/\*/.*/g ; s/\(.*\)/\^\1\$/g' "$globs") | tee -a "$files_out"
```

Comment:
I think this kind of "glob to regex" transform is going to be buggy and arcane. I had a google around and found a cleaner solution:
```
cat "$globs" | while read -r pattern; do
  [ -z "$pattern" ] && continue # skip blank lines
  [[ "$pattern" == "#*" ]] && continue # skip commented lines
  if [[ "$filename" == "$pattern" ]]; then
    echo "$filename"
    break
  fi
done
```
This uses the property of `[[`'s `==` operator, where the RHS is interpreted as a glob pattern.